File: src/main/java/com/travel/assistant/modules/activity/entity/Activity.java
package com.travel.assistant.modules.activity.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.travel.assistant.entity.BaseEntity;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 活动实体类
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("t_activity")
public class Activity extends BaseEntity {

    /**
     * 创建者ID
     */
    private Long creatorId;

    /**
     * 活动标题
     */
    private String title;

    /**
     * 活动描述
     */
    private String description;

    /**
     * 活动地点
     */
    private String location;

    /**
     * 经度
     */
    private BigDecimal longitude;

    /**
     * 纬度
     */
    private BigDecimal latitude;

    /**
     * 开始时间
     */
    private LocalDateTime startTime;

    /**
     * 结束时间
     */
    private LocalDateTime endTime;

    /**
     * 最大参与人数
     */
    private Integer maxParticipants;

    /**
     * 当前参与人数
     */
    private Integer currentParticipants;

    /**
     * 状态：0已取消 1进行中 2已结束
     */
    private Integer status;
}

File: src/types/activity.ts
// 活动状态枚举
export enum ActivityStatus {
  PENDING = 0,
  ONGOING = 1,
  COMPLETED = 2,
  CANCELLED = 3,
}

// 活动信息
export interface ActivityVO {
  id: number
  creatorId: number
  creatorNickname: string
  title: string
  description: string
  location: string
  longitude: number
  latitude: number
  startTime: string
  endTime: string
  maxParticipants: number
  currentParticipants: number
  status: ActivityStatus
  statusDesc: string
  joined: boolean
  createdAt: string
}

// 活动请求
export interface ActivityRequest {
  title: string
  description?: string
  location?: string
  longitude?: number
  latitude?: number
  startTime?: string
  endTime?: string
  maxParticipants?: number
}

// 活动列表查询参数
export interface ActivityListParams {
  page?: number
  size?: number
  keyword?: string
  status?: ActivityStatus
}